### PR TITLE
vc profile page

### PR DESCRIPTION
**Pull request by reactoholic**
vc profile page

## Summary by CodeRabbit

- **New Features**
  - Enhanced localization for the Virtual Contributor, including clearer data privacy and external AI service integration details.
  - Added dynamic rendering in the VCProfilePageView based on the presence of body of knowledge ID.

- **Bug Fixes**
  - Simplified loading state management in VCProfilePage.

- **Documentation**
  - Updated translation keys for improved clarity and consistency.

- **Refactor**
  - Improved prop structure and conditional rendering logic in VCProfilePageView.

- **Chores**
  - Reorganized import statements across components for better readability.

### CODE REVIEW

File: .build/docker/.env.base [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-VITE_APP_BUILD_VERSION=dev
-VITE_APP_BUILD_DATE=dev
-VITE_APP_BUILD_REVISION=dev
-VITE_APP_ALKEMIO_DOMAIN=http://localhost:3000
-VITE_APP_SENTRY_AUTH_TOKEN=
-VITE_APP_LOG_ERRORS=true
-VITE_APP_COLLAB_URL=http://localhost:4002
-VITE_APP_COLLAB_PATH=/socket.io
-VITE_APP_GRAPHQL_ENDPOINT=/graphql
+VITE_APP_VERSION           = 0.75.2
```
Comment (bobbykolev) [inline .build/docker/.env.base:1]: Let's exclude this change. This file is in the .gitignore, and I'm not sure where you're getting this formatting.

File: src/core/i18n/en/translation.en.json [hunk 2]
```diff
@@ -2621,6 +2621,15 @@
...
+            "description": "This Virtual Contributor operates using the following external AI service: <strong>OpenAI</strong>. All questions submitted to this Virtual Contributor, along with any related comments in a threaded conversation, will be sent to <strong>OpenAI</strong>. Please be aware that this data may be used to train and improve their model."
+          },
+          "privacy": {
+            "description": "At Alkemio, protecting your data is our top priority. We are fully committed to ensuring the security and privacy of your information. Here’s how we handle your data when interacting with a Virtual Contributor (VC):",
+            "bullets": "<ul><li>Only the content of your question and any related comments in a threaded conversation are shared with the AI service that powers the Virtual Contributor.</li><li>Please be aware that if the Virtual Contributor is based on an external AI service, such as , the data shared may be used to further train and improve that service.</li><li>Additionally, note that questions and answers exchanged with the Virtual Contributor may be accessible to the Alkemio team for monitoring or support purposes.</li></ul>"
```
[the remark this answers is not in the document]
Comment (bobbykolev) [reply src/core/i18n/en/translation.en.json:2631]: It makes sense, please double check what should be places after `, such as `

File: .build/docker/.env.base (hunk 1 above)
Comment (reactoholic) [reply .build/docker/.env.base:1]: It's coming from an extension which I just removed from my IDE. If this file is in the `.gitignore` file how come git detects it as a change?

File: src/domain/community/virtualContributor/vcProfilePage/model.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { BasicSpaceProps } from '../components/BasicSpaceCard';
+
+export type VCProfilePageViewProps = {
```
Comment (bobbykolev) [inline src/domain/community/virtualContributor/vcProfilePage/model.ts:3]: I agree on extracting the types/interfaces in separate files. 
However, what would be the convention? What fi we have a couple of them like ProfileProps but also ProfileFormProps?
This `model` could be misleading and usually we're not putting every component in a separate folder.
It's not critical for this PR but let's decide on how we're going to approach it in the future together with @ccanos.

Comment (reactoholic) [reply src/domain/community/virtualContributor/vcProfilePage/model.ts:3]: The motivation behind the name `model.ts` is coming due the fact that this file lives in the `vcProfilePage` folder which means this `model.ts` is containing types for every file that lives in that `vcProfilePage` folder. The `model.ts` file can include many types as long as they are strictly related to the files in the `vcProfilePage` folder.

File: src/domain/community/virtualContributor/vcProfilePage/VCProfilePageView.tsx [hunk 4]
```diff
@@ -1,43 +1,21 @@
...
-      };
-    };
-  };
-  bokProfile?: BasicSpaceProps;
-  bokDescription?: string;
-}
+import { type VCProfilePageViewProps } from './model';
 
 const SectionTitle = ({ children }) => (
   <BlockTitle display={'flex'} alignItems={'center'} gap={theme => theme.spacing(1)}>
```
[the remark this answers is not in the document]
Comment (bobbykolev) [reply src/domain/community/virtualContributor/vcProfilePage/VCProfilePageView.tsx:21]: Please, see the usage of gap in other places. We either pass a value - 1 or we're using the gutters() util which is using the theme spacing.

File: src/core/i18n/en/translation.en.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (reactoholic) [reply src/core/i18n/en/translation.en.json:2631]: > It makes sense, please double check what should be places after `, such as `

I asked Simone about this, currently waiting for an answer. corerabbit's warning makes sense, though.

File: src/domain/community/virtualContributor/vcProfilePage/VCProfilePageView.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (reactoholic) [reply src/domain/community/virtualContributor/vcProfilePage/VCProfilePageView.tsx:21]: > Please, see the usage of gap in other places. We either pass a value - 1 or we're using the gutters() util which is using the theme spacing.

`gap={gutters(0.5)}` is the equivalent of `gap={theme => theme.spacing(1)}`, changing it right away!

**Review by bobbykolev — APPROVED** [review]
Nice! Thanks for fixing and improving some of the existing code as well!

The PR tags are also a nice touch, I'll start using them as well.

👏

**Pull request merged**